### PR TITLE
Add initial talk functionality

### PR DESCRIPTION
**Pull request by aweiksnar**
Here's a quick feature synopsis of where I'm at with talk right now:

**Working:** Comment/Board/Discussion create/edit/destroy, Markdown rendering, Upvotes, Project talk - main talk distinction, tags/user mentions, Talk auth , Pagination, Comment live preview, Comment/discussion validations

**Incomplete/TODO:** Featured Image, Avatar rendering, User roles/moderation, Copy, Feedback messaging after actions, Splitting up css

**Placeholders for the future:** Reply, Report, Comment Link, Search, Individual Subject links

Happy to re-organize/change things up if you notice anything looks odd or breaks convention

### CODE REVIEW

File: app/api/talk.coffee [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+apiClient = require './client'
+authClient = require './auth'
+
+# Staging:
+# https://talk-staging.zooniverse.org
+
+# Local:
+# http://127.0.0.1:3000
+
+talkClient = new JSONAPIClient 'https://talk-staging.zooniverse.org',
```
Comment (brian-c) [inline app/api/talk.coffee]: Can we define the possible Talk API hosts in api/config.coffee so it respects any set environment?

File: app/talk/board.cjsx [hunk 2]
```diff
@@ -0,0 +1,183 @@
...
+    @discussionsRequest(page)
+      .then (discussions) =>
+        discussionsMeta = discussions[0]?.getMeta()
+        @setState {discussions, discussionsMeta}
+      .catch (e) =>
+        console.log "failed to get discussion", e
+
+  boardRequest: ->
+    id = +@props.params.board
+    talkClient.type('boards').get({id})
```
Comment (brian-c) [inline app/talk/board.cjsx]: Resource IDs are always strings. `Type::get`, given a string, will resolve to that resource.

**Comment by brian-c** [comment]
Sweet, looks like a really nice base. I'm going to merge this in and as we iterate on it I might have more picky comments.

Thanks @aweiksnar!

**Comment by aweiksnar** [comment]
Awesome sounds good, I'll add in the changes you mentioned to the next PR

**Pull request merged**